frontend/hub/collections/UploadCollection.tsx

```diff
@@ -170,82 +162,70 @@ export function UploadCollectionByFile() {
     );
   }
 
+  function getNamespaceNameFromFile(file?: File) {
+    return file?.name.split('-')[0] ?? '';
+  }
+
   async function submitData(data: UploadData) {
-    const list = await distroGetRequest(
-      pulpAPI`/distributions/ansible/ansible/?repository=${selectedRepo?.pulp_href || ''}`
-    );
-    const base_path = list?.results[0]?.base_path;
+    setError('');
+    const namespaceName = getNamespaceNameFromFile(data.file as File);
+    if (namespaceParams && namespaceName !== namespaceParams) {
+      setError(
+        t(`Namespace "{{namespaceName}}" do not match namespace "{{namespaceParams}}"`, {
+          namespaceName,
+          namespaceParams,
+        })
+      );
+      return;
+    }
 
-    return hubPostRequestFile(
-      hubAPI`/v3/plugin/ansible/content/${base_path}/collections/artifacts/`,
-      data.file as Blob
-    ).then(() => pageNavigate(HubRoute.Approvals));
+    let lastError = '';
+
+    try {
+      lastError = t('Error in loading namespace {{namespaceName}}', { namespaceName });
+      const namespace = await requestGet<HubItemsResponse<HubNamespace>>(
```

Comment:
This error handle is very different from the other fields, in general we are either showing the error associated with a certain field, or displaying an error at the form level. The styling also does not fit with the remaining errors one. 

![image](https://github.com/ansible/ansible-ui/assets/9053044/537071ee-199b-4a42-84cb-a9cb0165c3fd)



If a HTTP request error is raising inside the HubPageForm it will be parsed to either be displayed at form level, or at field level. It might be worth to explore raising a RequestError and pass the proper strings there.

Reply:
However, this is not using PageForm... It was written before we started to do any PageForms and also the basics were written by James and he also did not used PageForm. Do we want to rewrite it? Mind that we dont have enough time. Some smaller fix to make it look more similar will be nicer.

Reply:
@MilanPospisil The submit is wrapped on the HubPageForm submit and any HTTP error raised inside that will bubble up and catch up by the HubPageForm - as long we do not catch the error inside the submit.

At least for the upload field itself we could do something like below, perhaps we could a function raiseFieldError, and raiseFormError to the framework, or as helpers.

I added  a function

 ```

async function raiseFieldError(fieldName: string, message: string, status = 404) {
  const responseBody = JSON.stringify({ [fieldName]: [message] });
  const response = new Response(responseBody, {
    status,
    statusText: httpStatusCodes[status],
    headers: { 'Content-Type': 'application/json' },
  });
  throw await createRequestError(response);
}

const httpStatusCodes: { [key: number]: string } = {
  200: 'OK',
  201: 'Created',
  204: 'No Content',
  400: 'Bad Request',
  401: 'Unauthorized',
  403: 'Forbidden',
  404: 'Not Found',
  500: 'Internal Server Error',
  502: 'Bad Gateway',
  503: 'Service Unavailable',
};
```

Then inside the onSubmit I added 

```

  if (namespace.data.length === 0) {
      await raiseFieldError(
        'file',
        t('Can not find namespace {{namespaceName}}', { namespaceName })
      );
    }

```


<img width="963" alt="image" src="https://github.com/ansible/ansible-ui/assets/9053044/92c3d48c-25e9-49d7-8145-f1e543ee7e37">

Reply:
You are right, I missed that it is using hub page form component.

Reply:
James forked https://github.com/ansible/ansible-ui/pull/1546